docs/reference/aggregations/bucket/categorize-text-aggregation.asciidoc

```diff
@@ -0,0 +1,462 @@
+[[search-aggregations-bucket-categorize-text-aggregation]]
+=== Categorize text aggregation
+++++
+<titleabbrev>Categorize text</titleabbrev>
+++++
+
+experimental::[]
+
+A multi-bucket aggregation that groups semi-structured text into buckets. Each `text` field is re-analyzed
+using a custom analyzer. The resulting tokens are then categorized creating buckets of similarly formatted
+text values. This aggregation works best with machine generated text like system logs.
+
+WARNING: Re-analyzing _large_ result sets will require a lot of time and memory. This aggregation should be
+         used in conjunction with <<async-search, Async search>>. Additionally, you may consider
+         using the aggregation as a child of either the <<search-aggregations-bucket-sampler-aggregation,sampler>> or
+         <<search-aggregations-bucket-diversified-sampler-aggregation,diversified sampler>> aggregation.
+         This will typically improve speed and memory use.
+
+NOTE: If you have considerable memory allocated to your JVM but are receiving circuit breaker exceptions from this
+      aggregation, you may be attempting to categorize text that is poorly formatted for categorization. Consider
+      adding `categorization_filters` or running under <<search-aggregations-bucket-sampler-aggregation,sampler>> or
+      <<search-aggregations-bucket-diversified-sampler-aggregation,diversified sampler>> to explore the created categories.
```

Comment:
> It'd be nice if there was some signal about which tokens were consuming the most memory. That's a "for later" thing, I think.

This would be pretty difficult. That would require some sort of pattern recognition inside of a given token position. This would be neat, but non-trivial.

Reply:
Yeah. Would you be ok moving this lower into the page?